packages/gatsby/src/query/query-compiler.js

```diff
@@ -503,7 +503,7 @@ const addExtraFields = (document, schema) => {
         if (node.name.value === `__typename`) {
```

Comment:
In theory this can also be aliased.

Reply:
I was pretty sure that field names starting with `__` are not allowed in GraphQL but apparently aliases are. Good catch, will add a check here as well.